fast_i18n_flutter/lib/fast_i18n_flutter.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:fast_i18n_dart/fast_i18n_dart.dart';
+import 'package:flutter/widgets.dart';
+
+extension ExtBaseAppLocale on BaseAppLocale {
+  Locale get flutterLocale {
+    return Locale.fromSubtags(
+      languageCode: languageCode,
+      scriptCode: scriptCode,
+      countryCode: countryCode,
+    );
+  }
+}
+
+final _translationProviderKey = GlobalKey<_TranslationProviderState>();
+
+class TranslationProvider extends StatefulWidget {
+  TranslationProvider({required this.child}) : super(key: _translationProviderKey);
+
+  final Widget child;
+
+  @override
+  _TranslationProviderState createState() => _TranslationProviderState();
+
+  static _InheritedLocaleData of(BuildContext context) {
+    final inheritedWidget = context.dependOnInheritedWidgetOfExactType<_InheritedLocaleData>();
+    if (inheritedWidget == null) {
+      throw 'Please wrap your app with "TranslationProvider".';
+    }
+    return inheritedWidget;
+  }
+}
+
+class _TranslationProviderState extends State<TranslationProvider> {
+  BaseAppLocale locale = _currLocale;
+
+  void setLocale(BaseAppLocale newLocale) {
+    setState(() {
+      locale = newLocale;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _InheritedLocaleData(
+      locale: locale,
+      child: widget.child,
+    );
+  }
+}
+
+class _InheritedLocaleData extends InheritedWidget {
+  final BaseAppLocale locale;
+  Locale get flutterLocale => locale.flutterLocale; // shortcut
+  final _StringsZh translations; // store translations to avoid switch call
```

Comment:
This is difficult to abstract :)

Reply:
Already done (with some modifications though)